File: backend/models/Quiz.js
const mongoose = require('mongoose');

const questionSchema = new mongoose.Schema({
  question: { type: String, required: true },
  options: [String],
  correctAnswer: { type: String, required: true },
  explanation: String
});

const quizSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: String,
  category: {
    type: String,
    required: true,
    enum: ['grammar', 'vocabulary', 'pronunciation', 'reading']
  },
  questions: [questionSchema],
  timeLimit: Number, // in minutes
  points: { type: Number, default: 10 },
  difficulty: {
    type: String,
    enum: ['easy', 'medium', 'hard'],
    default: 'easy'
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Quiz', quizSchema); 

File: frontend/lib/models/quiz.dart
// class Quiz {
//   final String question;
//   final List<String> options;
//   final int answerIndex;
//   final String category;
//   final int level;
//
//   Quiz({
//     required this.question,
//     required this.options,
//     required this.answerIndex,
//     required this.category,
//     required this.level,
//   });
//
//   factory Quiz.fromJson(Map<String, dynamic> json) => Quiz(
//     question: json['question'],
//     options: List<String>.from(json['options']),
//     answerIndex: json['answerIndex'],
//     category: json['category'],
//     level: json['level'],
//   );
// }
//

class Quiz {
  final String question;
  final List<String> options;
  final int answerIndex;
  final String category;
  final int level;

  Quiz({
    required this.question,
    required this.options,
    required this.answerIndex,
    required this.category,
    required this.level,
  });

  factory Quiz.fromJson(Map<String, dynamic> json) => Quiz(
    question: json['question'],
    options: List<String>.from(json['options']),
    answerIndex: json['answerIndex'],
    category: json['category'],
    level: json['level'],
  );

  Map<String, dynamic> toJson() => {
    'question': question,
    'options': options,
    'answerIndex': answerIndex,
    'category': category,
    'level': level,
  };
}